src/Utils/CallbackRouteGenerator.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the DocusignBundle.
+ *
+ * (c) Grégoire Hébert <gregoire@les-tilleuls.coop>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace DocusignBundle\Utils;
+
+use DocusignBundle\EnvelopeBuilderInterface;
+use Symfony\Component\Routing\Router;
+use Symfony\Component\Routing\RouterInterface;
+
+class CallbackRouteGenerator
+{
+    public static function getCallbackRoute(RouterInterface $router, EnvelopeBuilderInterface $envelopeBuilder): string
+    {
+        $queryParameters = array_unique(['envelopeId' => $envelopeBuilder->getEnvelopeId()] + $envelopeBuilder->getCallbackParameters());
+
+        // Callback is a route name
+        if (!preg_match('/^https?:\/\//', $envelopeBuilder->getCallback())) {
```

Comment:
[Idea] We could always handle a route name for the callback, even for an external url. WDYT?
```yaml
docusign_callback_foo:
    path: /docusign/callback/foo
    defaults:
        _controller: FrameworkBundle:Redirect:urlRedirect
        path: "http://example.com/blog"
        permanent: true
```
This could be done in the [routing loader](https://github.com/GregoireHebert/docusign-bundle/pull/35).